### PR TITLE
Multi bookie cluster Support

### LINKED ISSUE #387: multi-bookie cluster support
Does not support multi-bookie cluster support

### PR DESCRIPTION
**Pull request by g0715158**
Fixes #387

Master Issue: #387

*Describe the modifications you've done.*

Support multi-bookie cluster, no longer use the address configured in the application.properties file

### CODE REVIEW

File: front-end/src/views/management/environments/index.vue [hunk 1]
```diff
@@ -76,183 +90,190 @@
...
-        if (!response.data) return
-        if (response.data.hasOwnProperty('error')) {
+      },
+      updateEnvironment() {
+        const data = {
+          'name': this.form.environment,
+          'broker': this.form.broker,
+          'bookie': this.form.bookie
+        }
+        updateEnvironment(data).then(response => {
```
Comment (urfreespace) [inline front-end/src/views/management/environments/index.vue:248]: 建议是用async/await @g0715158

**Review by urfreespace — COMMENTED** [review]
LGTM

**Review by urfreespace — APPROVED** [review]

File: front-end/src/views/management/environments/index.vue (hunk 1 above)
Comment (g0715158) [reply front-end/src/views/management/environments/index.vue:248]: 保持和原来的实现方式，我添加了一个字段

Comment (urfreespace) [reply front-end/src/views/management/environments/index.vue:248]: OK

File: src/main/java/org/apache/pulsar/manager/service/impl/BookiesServiceImpl.java [hunk 2]
```diff
@@ -33,26 +35,35 @@
...
+    private final HttpServletRequest request;
     @Value("${backend.directRequestBroker}")
     private boolean directRequestBroker;
 
     @Value("${backend.directRequestHost}")
     private String directRequestHost;
 
-    @Value("${bookie.host}")
-    private String bookieHost;
+//    @Value("${bookie.host}")
```
Comment (eolivelli) [inline src/main/java/org/apache/pulsar/manager/service/impl/BookiesServiceImpl.java:46]: Please remove commented code

**Review by eolivelli — APPROVED** [review]
Lgtm

Thanks for contributing this feature

**Review by urfreespace — APPROVED** [review]

File: src/main/java/org/apache/pulsar/manager/service/impl/BookiesServiceImpl.java (hunk 2 above)
Comment (g0715158) [reply src/main/java/org/apache/pulsar/manager/service/impl/BookiesServiceImpl.java]: @eolivelli 
The commented code has been deleted

Comment (eolivelli) [reply src/main/java/org/apache/pulsar/manager/service/impl/BookiesServiceImpl.java]: thanks

File: front-end/src/views/management/environments/index.vue (hunk 1 above)
Comment (tuteng) [reply front-end/src/views/management/environments/index.vue:248]: comment It is recommended to use English so that all contributors can read it

**Review by tuteng — APPROVED** [review]

File: front-end/src/views/management/environments/index.vue (hunk 1 above)
Comment (tuteng) [reply front-end/src/views/management/environments/index.vue:248]: > 保持和原来的实现方式，我添加了一个字段

Translation
To keep it the same as the original implementation, I added a field

**Comment by tuteng** [comment]
Test failed: https://github.com/apache/pulsar-manager/pull/388/checks?check_run_id=2654368792

image

**Review by urfreespace — APPROVED** [review]

**Review by urfreespace — APPROVED** [review]

**Review by urfreespace — APPROVED** [review]

**Pull request merged**